### ISSUE TITLE
Separate ACK handler

### ISSUE THREAD

**Issue by vuong177** [notional-labs/fee-abstraction#36]
We have two separate ACK: query price ack and swap token ack. We need to separate it in OnAcknowledgementPacket.

**Comment by GNaD13** [comment]
Hey, I already setup for this in #30,can you check it?

**Comment by GNaD13** [comment]
> We have two separate ACK: query price ack and swap token ack. We need to separate it in OnAcknowledgementPacket.

No need now because we can use osmosis cross-chain swap instead

**Comment by vuong177** [comment]
yes

**Comment by GNaD13** [comment]
reopen because now we need to handle swap on axelar, there are 2 choice:
- using feeabs ibc msg
- ICA Controller msg

**Comment by GNaD13** [comment]
@expertdicer I think we need to check enum msg on contract so we can handle these type of msg

**Comment by expertdicer** [comment]
> @expertdicer I think we need to check enum msg on contract so we can handle these type of msg

Yes. I have made some works on enum msg with one of protobuf. Can you specify what kinds of msg ?
